include/boost/url/impl/any_allocator.hpp

```diff
@@ -0,0 +1,243 @@
+            case alignof(aligned_16_t):
+                return reinterpret_cast<char*>(aligned_16_allocator_type(a_).allocate(n * ceil_div(size, align)));
+            default:
+                return reinterpret_cast<char*>(aligned_max_allocator_type(a_).allocate(n * ceil_div(size, align)));
+            }
+        }
+
+        void
+        deallocate(char* p, std::size_t n, std::size_t size, std::size_t align) {
+            switch (align) {
+            case alignof(aligned_1_t): {
+                if (size == 1) {
+                    a_.deallocate(p, n);
+                } else {
+                    aligned_1_allocator_type(a_).deallocate(reinterpret_cast<aligned_1_t*>(p), n * ceil_div(size, align));
+                }
+            break;
+            }
+            case alignof(aligned_2_t):
+                aligned_2_allocator_type(a_).deallocate(reinterpret_cast<aligned_2_t*>(p), n * ceil_div(size, align));
+                break;
+            case alignof(aligned_4_t):
+                aligned_4_allocator_type(a_).deallocate(reinterpret_cast<aligned_4_t*>(p), n * ceil_div(size, align));
+                break;
+            case alignof(aligned_8_t):
+                aligned_8_allocator_type(a_).deallocate(reinterpret_cast<aligned_8_t*>(p), n * ceil_div(size, align));
+                break;
+            case alignof(aligned_16_t):
+                aligned_16_allocator_type(a_).deallocate(reinterpret_cast<aligned_16_t*>(p), n * ceil_div(size, align));
+                break;
+            default:
+                aligned_max_allocator_type(a_).deallocate(reinterpret_cast<std::max_align_t*>(p), n * ceil_div(size, align));
+                break;
+            }
+        }
+
+        void
+        destroy_self() noexcept override
+        {
+            deallocate(reinterpret_cast<char*>(this), 1, sizeof(impl), alignof(impl));
+        }
+
+        bool
+        is_always_equal() const {
+            return boost::allocator_is_always_equal_t<char_allocator_type>::value;
+        }
+    };
+
+    using impl_allocator_type = boost::allocator_rebind_t<Allocator, impl>;
+    impl_allocator_type al(a);
+    p_ = ::new(al.allocate(1)) impl(al);
+}
+
+template <class T>
+any_allocator<T>::
+any_allocator(
+    any_allocator const& other) noexcept : p_(other.p_)
+{
+    if(p_)
+        ++p_->refs;
+}
+
+template <class T>
+any_allocator<T>::
+any_allocator(
+    any_allocator&& other) noexcept : p_(std::move(other.p_))
+{
+    other.p_ = nullptr;
+}
+
+template <class T>
+template <class U>
+any_allocator<T>::
+any_allocator(
+    any_allocator<U> const& other) noexcept
+: p_(reinterpret_cast<any_allocator<T>::base*>(other.p_))
+{
+    if(p_)
+        ++p_->refs;
+}
+
+template <class T>
+template <class U>
+any_allocator<T>::
+any_allocator(
+    any_allocator<U> && other) noexcept
+: p_(reinterpret_cast<any_allocator<T>::base*>(other.p_))
```

Comment:
This is undefined behavior. You need to move the types `base` and `impl`  outside of `any_allocator`.

Reply:
That's in the todo list.